### PR TITLE
L2vpn/evpn support

### PR DESCRIPTION
**Pull request by saichint**
This PR is for extending afi/safi to include l2vpn/evpn support. This address family is now supported across n5k, n6k, n7k, n8k and n9k. 

The changes are demo_manifest and beaker tests. 
While tested the demo_manifest few other issues cropped up for bgp and bgp_neighbor_af providers and those are taken care as well.
For beaker, added new tests for title patterns to include l2vpn/evpn. 
This runs for n5k, n6k, n7k, n8k and n9k. 
For n9k, it is supported only for I3 images and so I2 is excluded.

CHANGELOG and README are also updated. For this PR, N8k support is not added in README as it needs to be added across all providers globally.

All beaker and demo manifest tests pas across all nexus platforms (except n3k)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [1.3.2] - 2016-05-12
```
Comment (mikewiebe) [inline CHANGELOG.md]: Please use the CHANGELOG in master as your starting point and add the 1.3.2 release info to that.

https://github.com/cisco/cisco-network-puppet-module/blob/master/CHANGELOG.md

File: README.md [hunk 2]
```diff
@@ -1052,15 +1052,18 @@ Manages configuration of a BGP Address-family instance.
 |----------|:------------------:|:----------------------:|
 | N9k      | 7.0(3)I2(1)        | 1.1.0                  |
 | N3k      | 7.0(3)I2(1)        | 1.1.0                  |
+| N5k      | 7.3(0)N1(1)        | 1.3.2                  |
```
Comment (mikewiebe) [inline README.md]: This should be `1.2.0` right?  We missed this.

File: README.md [hunk 3]
```diff
@@ -1052,15 +1052,18 @@ Manages configuration of a BGP Address-family instance.
...
 | N6k      | 7.3(0)N1(1)        | 1.2.0                  |
 | N7k      | 7.3(0)D1(1)        | 1.2.0                  |
 
 #### <a name="cisco_bgp_af-caveats">Caveats</a>
 
 | Property | Caveat Description |
 |:--------|:-------------|
-| `additional_paths_install` | Not supported on N3k, N9k |
-| `advertise_l2vpn_evpn`     | Not supported on N3k, N6k |
```
Comment (mikewiebe) [inline README.md]: How about this instead?

| Property | Caveat Description |
|:------------|:--------------------------|
| `additional_paths_install` | Not supported on N3k, N9k|
| `advertise_l2vpn_evpn`     | Not supported on N3k, N6k|
| `afi - l2vpn`| Module Minimum Version `1.3.2`<br>OS Minimum Version `7.0(3)I3(1)`<br>Not supported on N3k |
| `safi - evpn`| Module Minimum Version `1.3.2`<br>OS Minimum Version `7.0(3)I3(1)`<br>Not supported on N3k |

File: README.md (hunk 2 above)
Comment (saichint) [inline README.md]: I thought so but was not very sure. so wanted to get your comment anyway.

File: README.md (hunk 3 above)
Comment (saichint) [inline README.md]: Yeah that looks better. done

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,12 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
-## [1.3.0] - 2016-05-02
+## [1.3.2] - 2016-05-12
+
+### Added
+- Extended `cisco_bgp_af` to include l2vpn/evpn support
```
Comment (chrisvanheuveln) [inline CHANGELOG.md]: This is really just adding it for AF support so:

`+- Extended `cisco_bgp_af` to include l2vpn/evpn address-family support `

File: README.md (hunk 3 above)
Comment (chrisvanheuveln) [inline README.md]: Again, I would clarify it with AF, and I would pair them:

| Property | Caveat Description |
|:------------|:--------------------------|
| `additional_paths_install` | Not supported on N3k, N9k|
| `advertise_l2vpn_evpn`     | Not supported on N3k, N6k|
| address-family `l2vpn/evpn`| Module Minimum Version `1.3.2`<br>OS Minimum Version `7.0(3)I3(1)`<br>Not supported on N3k |

File: examples/cisco/demo_bgp.pp [hunk 5]
```diff
@@ -188,6 +187,27 @@
     redistribute                  => $ipv4_redistribute,
   }
 
+  cisco_bgp_af { '55.77 default l2vpn evpn':
```
Comment (chrisvanheuveln) [inline examples/cisco/demo_bgp.pp]: This will break on 3k right? ie. needs a platform check.

File: tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb [hunk 6]
```diff
@@ -185,6 +185,35 @@
   resource:      { 'ensure' => 'present' },
 }
 
+tests[:title_patterns_4] = {
```
Comment (chrisvanheuveln) [inline tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb]: IMO these new title_pattern tests are not needed. All they really do is test the title regex logic and that's already being done by T.1 - T.3.

What I would do for beaker is just add a basic test a la:

```
tests[:l2vpn_evpn] = {
   desc:           '3.1 Address-family l2evpn/evpn',
   platform:         'n(5|6|7|8|9)k',
   title_pattern:  '2 default l2vpn evpn',
   manifest_props: {
     ensure: 'present',
  }
}

test_harness_run(tests, :l2vpn_evpn) unless nexus_i2_image

```

**Comment by saichint** [comment]
Made all the changes. Rested on n3k (I2, I3), n6k, n7k, n9k. All demo_manifest and beaker tests pass.

**Pull request merged**